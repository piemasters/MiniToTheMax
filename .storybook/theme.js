import { create } from "@storybook/theming/create";
import logo from "../content/assets/images/icon.png";


const theme = create({
    base: "light",

    colorPrimary: "#ff5470",
    colorSecondary: "#00dcb5",

    // UI
    appBg: "white",
    appContentBg: "white",
    appBorderColor: "grey",
    appBorderRadius: 10,

    // Typography
    fontBase: '"Open Sans", sans-serif',
    fontCode: "monospace",

    // Text colors
    textColor: "black",
    textInverseColor: "rgba(255,255,255,0.9)",

    // Toolbar default and active colors
    barTextColor: "white",
    barSelectedColor: "#00dcb5",
    barBg: "#00153d",

    // Form colors
    inputBg: "white",
    inputBorder: "white",
    inputTextColor: "black",
    inputBorderRadius: 4,

    brandTitle: "MiniToTheMax Storybook",
    brandUrl: "https://minitothemax.app",
    brandImage: logo
});

export default theme;