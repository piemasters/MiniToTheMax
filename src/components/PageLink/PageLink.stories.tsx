import type { Meta, StoryObj } from '@storybook/react';

import { PageLink } from './PageLink';

const meta: Meta<typeof PageLink> = {
  component: PageLink,
  argTypes: {
    type: {
      name: 'type',
      description: 'The animation type',
      defaultValue: 'paintDrip',
      options: ['paintDrip', 'fade', 'swipe', 'cover'],
      control: {
        type: 'select',
      },
    },
    to: {
      name: 'to',
      description: 'The internal link address',
      defaultValue: '/',
      control: { type: 'text' },
    },
    direction: {
      name: 'direction',
      description: 'Directional of transitions',
      defaultValue: 'up',
      options: ['left', 'right', 'up', 'down'],
      control: { type: 'select' },
    },
    children: {
      name: 'children',
      description: 'The html wrapped inside the link',
      defaultValue: 'Test',
      control: { type: 'text' },
    },
    linkActiveStyle: {
      name: 'linkActiveStyle',
      description:
        'The styling to apply to the link and its contents when active',
      // defaultValue: css`
      //   color: #000000;
      // `,
      control: { type: 'object' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof PageLink>;

export const Primary: Story = {
  args: {
    type: 'paintDrip',
    to: '/',
    direction: 'up',
    children: 'Test',
    linkActiveStyle: 'text-red-500!',
  },
};
