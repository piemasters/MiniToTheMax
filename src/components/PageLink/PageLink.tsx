import React, { FC } from 'react';
import { Link } from 'gatsby-link';

type AnimationTypes = 'paintDrip' | 'fade' | 'swipe' | 'cover';

export interface PageLinkProps {
  children: React.ReactNode;
  type?: AnimationTypes;
  to: string;
  linkActiveStyle?: string;
  direction?: string;
  disabled?: boolean;
  className?: string;
}

export const PageLink: FC<PageLinkProps> = ({
  children,
  type,
  to,
  linkActiveStyle,
  direction,
  disabled,
  className,
}) => {
  const baseClasses =
    className ||
    'text-blue-500 visited:text-blue-500 hover:text-blue-700 no-underline';

  if (disabled) {
    return <div className="text-gray-700">{children}</div>;
  }

  return (
    <Link
      data-testid="page-link"
      to={to}
      className={baseClasses}
      activeClassName={linkActiveStyle}
      partiallyActive={to !== '/'}
    >
      {children}
    </Link>
  );

  // if (type === 'paintDrip') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-paint"
  //       paintDrip
  //       hex={hex ? hex : '#eb1d23'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'fade') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-fade"
  //       fade={type === 'fade'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'swipe') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-swipe"
  //       swipe
  //       direction={direction ? direction : 'up'}
  //       top={top ? top : 'exit'}
  //       entryOffset={entryOffset ? entryOffset : 100}
  //       hex={hex ? hex : '#eb1d23'}
  //       to={to}
  //       duration={duration}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else if (type === 'cover') {
  //   return (
  //     <AniLink
  //       data-testid="page-link-cover"
  //       cover
  //       to={to}
  //       direction={direction ? direction : 'up'}
  //       bg={bg ? bg : '#eb1d23'}
  //       duration={duration ? duration : 1}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </AniLink>
  //   );
  // } else {
  //   return (
  //     <TransitionLink
  //       data-testid="page-link"
  //       to={to}
  //       css={activeStyle}
  //       activeStyle={linkActiveStyle}
  //       partiallyActive={to !== '/'}
  //     >
  //       {children}
  //     </TransitionLink>
  //   );
  // }
};

export default PageLink;
