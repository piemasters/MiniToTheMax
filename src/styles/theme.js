export default {
  colors: {
    text: '#000',
    muted: '#999',
  },
};
