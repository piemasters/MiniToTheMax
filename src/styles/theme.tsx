export interface Theme {
  colors: {
    text: string;
    textSecondary: string;
    textLight: string;
    hyperlink: string;
    hyperlinkActive: string;
    muted: string;
    lightgrey: string;
    lightgreyHover: string;
    primary: string;
    backgroundTransparent: string;
  };
}

const appTheme: Theme = {
  colors: {
    text: '#000',
    textSecondary: '#666',
    textLight: 'rgba(255, 255, 255, 0.9)',
    hyperlink: '#4f61ee',
    hyperlinkActive: '#0231e9',
    muted: '#999',
    lightgrey: '#ececec',
    lightgreyHover: 'e4e4e4',
    primary: '#eb1d23',
    backgroundTransparent: 'rgba(0, 0, 0, 0.7)',
  },
};

export default appTheme;
