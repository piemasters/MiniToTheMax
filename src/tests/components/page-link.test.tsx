import React from 'react';
import PageLink from '../../components/page-link';
import { render, getByTestId } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../styles/theme';

describe('PageLink', () => {
  test('renders correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageLink to={'/'}>Default</PageLink>
      </ThemeProvider>
    );

    // expect(getByTestId(container, 'page-link')).toHaveTextContent('Default');
    // expect(container).toMatchSnapshot();
  });

  test('paintDrip renders correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageLink type={'paintDrip'} to={'/'}>
          Paint Drip
        </PageLink>
      </ThemeProvider>
    );

    // expect(getByTestId(container, 'page-link-paint')).toHaveTextContent(
    //   'Paint Drip'
    // );
    // expect(container).toMatchSnapshot();
  });

  test('fade renders correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageLink type={'fade'} to={'/'}>
          Fade
        </PageLink>
      </ThemeProvider>
    );

    // expect(getByTestId(container, 'page-link-fade')).toHaveTextContent('Fade');
    // expect(container).toMatchSnapshot();
  });

  test('swipe renders correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageLink type={'swipe'} to={'/'}>
          Swipe
        </PageLink>
      </ThemeProvider>
    );

    // expect(getByTestId(container, 'page-link-swipe')).toHaveTextContent(
    //   'Swipe'
    // );
    // expect(container).toMatchSnapshot();
  });

  test('cover renders correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageLink type={'cover'} to={'/'}>
          Cover
        </PageLink>
      </ThemeProvider>
    );

    // expect(getByTestId(container, 'page-link-cover')).toHaveTextContent(
    //   'Cover'
    // );
    // expect(container).toMatchSnapshot();
  });
});
