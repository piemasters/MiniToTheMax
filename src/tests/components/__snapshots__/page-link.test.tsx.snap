// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink cover renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    bg="#eb1d23"
    class=""
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="page-link-cover"
    direction="up"
    duration="1"
    href="/"
  >
    Cover
  </a>
</div>
`;

exports[`PageLink fade renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    class=""
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="page-link-fade"
    href="/"
  >
    Fade
  </a>
</div>
`;

exports[`PageLink paintDrip renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    class=""
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="page-link-paint"
    hex="#eb1d23"
    href="/"
  >
    Paint Drip
  </a>
</div>
`;

exports[`PageLink renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    class=""
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="page-link"
    href="/"
  >
    Default
  </a>
</div>
`;

exports[`PageLink swipe renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    class=""
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="page-link-swipe"
    direction="up"
    hex="#eb1d23"
    href="/"
    top="exit"
  >
    Swipe
  </a>
</div>
`;
